packages/monitors/src/SyntheticPegMonitor.js

```diff
@@ -237,14 +241,17 @@ class SyntheticPegMonitor {
     }
   }
 
-  // Return historical volatility for pricefeed over specified time range and latest price.
+  // Return historical volatility for pricefeed over specified time range and latest price,
+  // or returns the timestamp from which historical price cannot be found.
   async _checkPricefeedVolatility(pricefeed) {
     // Get all historical prices from `volatilityWindow` seconds before the last update time and
     // record the minimum and maximum.
     const latestTime = pricefeed.getLastUpdateTime();
     const volData = this._calculateHistoricalVolatility(pricefeed, latestTime, this.volatilityWindow);
     if (!volData) {
-      return null;
+      return {
+        latestTime: latestTime ? latestTime : 0 // Return this for error logging purposes
```

Comment:
Question to all: is 0 a reasonable default value? Otherwise its possible that `pricefeed.getLastUpdateTime()` returns `null`, but I think this is a bit more readable.

Reply:
0 seems fine to me, given you can somehow make the var name a bit more descriptive.

Reply:
not suggesting a change here necessarily since what you have is a common pattern in this code base.  But, IMO these types of functions where success is "optional", that opinion should be made by the caller, or as further back in the callstack as possible. So in this case im implying that if there is no data, it should be considered an error in the called function and left to the caller to continue if its not catastrophic.  

Since the caller doesnt may not have the full context of why the error is generated, this would be a good candidate for a logging either in line or as an "event" where the internal function could "emit" with its context and the type of message and data you want, to be handled and logged in some parent module.

Just want to illustrate how this could be handled a bit more obviously to someone looking at the code as food for thought.

Reply:
>  So in this case im implying that if there is no data, it should be considered an error in the called function and left to the caller to continue if its not catastrophic.

I 100% agree. As you noted however this is unfortunately a pattern throughout this file and I didn't want to introduce a lot of code churn. Something we should consider for the Perpetual bots?